Framework/Crystal/src/NormaliseVanadium.cpp

```diff
@@ -74,15 +74,15 @@ void NormaliseVanadium::exec() {
     //    PARALLEL_START_INTERUPT_REGION //FIXME: Restore
 
     // Get a reference to the Y's in the output WS for storing the factors
-    MantidVec &Y = correctionFactors->dataY(i);
-    MantidVec &E = correctionFactors->dataE(i);
+    auto &Y = correctionFactors->mutableY(i);
+    auto &E = correctionFactors->mutableE(i);
 
     // Copy over bin boundaries
     const auto &inSpec = m_inputWS->getSpectrum(i);
-    const MantidVec &Xin = inSpec.readX();
-    correctionFactors->dataX(i) = Xin;
-    const MantidVec &Yin = inSpec.readY();
-    const MantidVec &Ein = inSpec.readE();
+    const auto &Xin = inSpec.x();
```

Comment:
Use `auto timeflight = inSpec.points()` below, that gives you the bin centers (which is, I think all you need, i.e., `Xin` can be removed). When using `timeflight` below you then probably need to use `timeflight.rawData()` and `timeflight.mutableRawData()` when passing it around.

Line 119:
```
timeflight.push_back((isHist ? (0.5 * (Xin[j] + Xin[j + 1])) : Xin[j]));
```
can be removed, and probably also the `isHist` variable.